meilisearch-core/src/update/synonyms_update.rs

```diff
@@ -56,21 +56,21 @@ impl SynonymsAddition {
     }
 }
 
-pub fn push_synonyms_addition(
+pub fn push_synonyms_update(
     writer: &mut heed::RwTxn<UpdateT>,
     updates_store: store::Updates,
     updates_results_store: store::UpdatesResults,
     addition: BTreeMap<String, Vec<String>>,
 ) -> MResult<u64> {
     let last_update_id = next_update_id(writer, updates_store, updates_results_store)?;
 
-    let update = Update::synonyms_addition(addition);
+    let update = Update::synonyms_update(addition);
     updates_store.put_update(writer, last_update_id, &update)?;
 
     Ok(last_update_id)
 }
 
-pub fn apply_synonyms_addition(
+pub fn apply_synonyms_update(
     writer: &mut heed::RwTxn<MainT>,
     main_store: store::Main,
     synonyms_store: store::Synonyms,
```

Comment:
You probably forget to `synonyms_store.clear()` before adding the new ones.

Reply:
The synonyms fst is just the list of all words that have alternatives, I am talking about the LMDB database (the kv store) that stores all the alternatives has the value of the key (the word that have alternatives).